### DISCUSSION TITLE
windows 11 24h2

### DISCUSSION THREAD

**Discussion by superrootplus** [AveYo/MediaCreationTool.bat#383] [Q&A]
windows 11 24h2?

**Comment by geissbuehler** [comment]
lzw29107 has added support for Win11 24H2 to his fork https://github.com/lzw29107/MediaCreationTool.bat. Unfortunately the installation fails on unsupported hardware, but fortunately I was able to enable the upgrade from Windows 11 23H2 to 24H2 with the registry modifications according to the [rufus FAQ](https://github.com/pbatard/rufus/wiki/FAQ#user-content-Windows_setup_says_my_PC_doesnt_meet_Windows_11_system_requirements).

I'm hoping that this registry modifications can be added to MediaCreationTool.bat to re-enable the installation of 24H2 on unsupported hardware. My understanding is that this will also work for upgrades from Windows 10 22H2 to Windows 11 24H2. Can someone test this upgrade on unsupported hardware?

**Reply by lacelestc** [reply]
Hello Yes Itested the fork modified script worked great for me to upgrade an unsupported hardware PC from w11 21H2 TO 24H2 but not directly .
look at modified script written by lzw29107 https://github.com/lzw29107/MediaCreationTool.bat
I had the blue screen crash so i figured out that my sdds drivers arnt compatible i disabled all of them except c where i put the iso and it s ok

**Reply by geissbuehler** [reply]
I have now performed more detailed tests with virtual machines running Windows 10 22H2 and Windows 11 23H2 using https://github.com/lzw29107/MediaCreationTool.bat/blob/main/MediaCreationTool.bat. Without the Rufus registry edits, on both machines the Windows 11 Setup was blocking the update because the PC does not meet the system requirements:

![Screenshot 2024-10-14 155709](https://github.com/user-attachments/assets/7c24bc82-6587-444c-813b-5fc153ad720a)

Unfortunately, the registry commands have been removed from the Rufus FAQ, but they are still preserved [here](https://www.neowin.net/news/rufus-posts-unsupported-cpu-bypass-for-windows-11-24h2-in-place-upgrade-but-not-for-everyone/), together with an explanation. I also pasted the commands below.

I assume that the 24H2 update via MediaCreationTool.bat fails because the `/product server` trick has been blocked by Microsoft. AveYo might need to include the registry modifications to reenable the 24H2 installation on unsupported hardware?

```
reg.exe delete "HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\CompatMarkers" /f 2>NUL
reg.exe delete "HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\Shared" /f 2>NUL
reg.exe delete "HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\TargetVersionUpgradeExperienceIndicators" /f 2>NUL
reg.exe add "HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\HwReqChk" /f /v HwReqChkVars /t REG_MULTI_SZ /s , /d "SQ_SecureBootCapable=TRUE,SQ_SecureBootEnabled=TRUE,SQ_TpmVersion=2,SQ_RamMB=8192,"
reg.exe add "HKLM\SYSTEM\Setup\MoSetup" /f /v AllowUpgradesWithUnsupportedTPMOrCPU /t REG_DWORD /d 1
```

**Comment by bdog549** [comment]
Just a FYI for anyone that sees this. I've used this bat file the last few years to go from 21H2 to 22H2 to 23H2 and just now upgraded to 24H2 successfully. As soon as I mounted the ISO, and ran the setup.exe, the script ran via a command prompt that opened for a few seconds and the install process resumed and it worked as intended. 

I can't recall which file I ran three years ago, either the auto.cmd or the ship TPM check on dynamic update. I'm guessing the latter.  If AveYo knows or anyone else please chime in so I can pass on the info. I just can't remember what I ran originally lol.

**Comment by khadzic** [comment]
To follow up on in this thread, is 24H2 in the pipeline to be added to the MediaCreatioTool?

**Comment by akaloith** [comment]
have clean installed dev 25h2 (rufus etc) on unsupported machine but cant update to latest dev
any help?

**Reply by poertyD** [reply]
Use ccleaner pro on a 60 free period.
Updates your drivers from Ccleaner.
There is about 14 drivers updates, 13 optional updates.
Then it should work.
I Cracked Win 11 Pro.
Installed the 25H2 update.
Works perfectly.

https://www.catalog.update.microsoft.com/Search.aspx?q=Updates+for+24H2+Windows+11+Pro&scol=Title&sdir=asc

Search for - Updatrs for 25H2

**Comment by kubadzik7777-hash** [comment]
I searched for a solution and found that if you want to upgrade Windows 11 to the latest version on unsupported hardware, you can do it using Rufus. Download the latest Win11 25h2 image from the MS website, take a USB flash drive, run Rufus. While creating the bootable USB flash drive, it will ask you if you want to use additional options, such as installing Win11 on unsupported hardware. Select what you need and create the USB flash drive. Then copy the contents of the USB flash drive to the machine you want to upgrade and run setup.exe (Run as Administrator). The installation will start, and within an hour, you'll be enjoying the new version of Win11 on unsupported hardware.
<img width="627" height="312" alt="Bez tytułu" src="https://github.com/user-attachments/assets/5c32db57-e51b-467f-ad16-f6020cfe1ed0" />
